x-pack/plugins/maps/public/shared/layers/styles/components/vector/legend/symbol_icon.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { getMakiSymbolSvg, styleSvg, buildSrcUrl } from '../../../symbol_utils';
+
+export class SymbolIcon extends Component {
+
+  state = {
+    imgDataUrl: undefined,
+    prevSymbolId: undefined,
+    prevFill: undefined,
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._loadSymbol(this.props.symbolId, this.props.fill);
+  }
+
+  componentDidUpdate() {
+    this._loadSymbol(this.props.symbolId, this.props.fill);
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async _loadSymbol(nextSymbolId, nextFill) {
+    if (nextSymbolId === this.state.prevSymbolId
+      && nextFill === this.state.prevFill) {
+      return;
+    }
+
+    let imgDataUrl;
+    try {
+      const svg = getMakiSymbolSvg(nextSymbolId);
+      const styledSvg = await styleSvg(svg, nextFill);
+      imgDataUrl = buildSrcUrl(styledSvg);
+    } catch (error) {
+      // ignore failures - component will just not display an icon
+      console.log(error);
+    }
+
+    if (this._isMounted) {
```

Comment:
Should we move this condition up to be a part of the top if-condition in this function and just return if `!this._isMounted`?

Reply:
Needs to be after checked after async action. After `await`, component may have been unmounted